packages/js-drive/lib/identity/masternode/createRewardShareDocumentFactory.js

```diff
@@ -41,6 +43,24 @@ function createRewardShareDocumentFactory(
       return false;
     }
 
+    const fetchedDocumentsResult = await documentRepository.find(
```

Comment:
I would keep only one query and check both number of docs and `payToId` equals to an operator in one of them.